packages/@core.keystone/plugins/softDeleted.js

```diff
@@ -41,9 +42,6 @@ const softDeleted = ({ deletedAtField = 'deletedAt', newIdField = 'newId' } = {}
     fields[newIdField] = { ...newIdOptions }
 
     const newResolveInput = ({ resolvedData, existingItem }) => {
-        if (existingItem && existingItem[deletedAtField]) {
```

Comment:
you forgot to check double delete action

Reply:
I've created a test case for this lad. Check 

`user: can not soft delete object twice`